### ISSUE TITLE
Add group-level hooks and push rules (and default branch protection example)

### ISSUE THREAD

**Issue by gdubicki** [egnyte/gitlabform#142]
It's better to provide group-level configs whenever you want some settings to apply to all projects within a group or groups.

They:
* apply to all the new projects by default,
* are faster to process with this app as you do it in a single place,
* generate less internal clutter in your GitLab instance,

For hooks and push rules we would need to actually implement the API but it's most probably copy-paste from project-level versions.

For default branch protection it's a matter of providing an example in the config, see https://docs.gitlab.com/ee/api/groups.html#update-group , `default_branch_protection` attribute.

**Comment by lfvjimisola** [comment]
Additional info from @gdubicki :

you can probably use the SingleEntityProcessor and add it with just a few lines of code, like in 

https://github.com/gdubicki/gitlabform/blob/fc7a3de0f8d16152257553c85b95d3077e9d2ee1/gitlabform/processors/project/project_push_rules_processor.py#L5-L13

```
class ProjectPushRulesProcessor(SingleEntityProcessor):
    def __init__(self, gitlab: GitLab):
        super().__init__(
            "project_push_rules",
            gitlab,
```

**Comment by lfvjimisola** [comment]
1. Will `enforce` be available in order to remove group level webhooks?
2. Is it just to set an empty string to empty a group push rule?

**Comment by amimas** [comment]
For group level webhooks, `enforce` should be supported along with `delete` key for individual webhook. They are supported for project level webhooks.

For "push rule", `enforce` most likely doesn't make sense. I think it's more appropriate to add support for `delete` key. It's not available in project level push rule config right now either.

> Is it just to set an empty string to empty a group push rule?

That would be the workaround right now.

**Comment by jimisola** [comment]
In the feature we should limits issues so that they are atomic in sense of features. This issue handles two features which I believe that one (group push rules) has now been implemented.

**Comment by amimas** [comment]
Agree. Implementations should be done as atomic as possible. It does require some administrative work from the maintainers too (i.e. grooming issues, labeling them, creating separate issues, enforcing it in incoming contributions without demotivating them, etc.). It can easily overwhelm you when you're spending all your personal time and effort into the project 😄 . But, I completely agree with you.
